system-contracts/contracts/libraries/EfficientCall.sol

```diff
@@ -36,7 +37,9 @@ library EfficientCall {
     /// @return The `keccak256` hash.
     function keccak(bytes calldata _data) internal view returns (bytes32) {
         bytes memory returnData = staticCall(gasleft(), KECCAK256_SYSTEM_CONTRACT, _data);
-        require(returnData.length == 32, "keccak256 returned invalid data");
+        if (returnData.length != 32) {
+            revert InvalidData();
```

Comment:
Confusing error, it should `Keccak256InvalidReturnData` or at least `InvalidReturnData`